kotest-framework/kotest-framework-engine/src/jvmMain/kotlin/io/kotest/engine/spec/ClassVisibilitySpecRefInterceptor.kt

```diff
@@ -0,0 +1,34 @@
+package io.kotest.engine.spec
+
+import io.kotest.core.internal.KotestEngineProperties
+import io.kotest.core.spec.SpecRef
+import io.kotest.core.test.TestCase
+import io.kotest.core.test.TestResult
+import io.kotest.engine.interceptors.EngineContext
+import io.kotest.engine.spec.interceptor.SpecRefInterceptor
+import kotlin.reflect.KVisibility
+
+/**
+ * A [SpecRefInterceptor] which will ignore private specs unless the include private flag
+ * is true in project config.
+ */
+class ClassVisibilitySpecRefInterceptor(private val context: EngineContext) : SpecRefInterceptor {
+
+   override suspend fun intercept(
+      ref: SpecRef,
+      fn: suspend (SpecRef) -> Result<Map<TestCase, TestResult>>
+   ): Result<Map<TestCase, TestResult>> {
+      return when {
+         ref.kclass.visibility == KVisibility.PRIVATE && !allowPrivate() -> Result.success(emptyMap())
+         else -> fn(ref)
+      }
+   }
+
+   /**
+    * We allow private classes if the configuration flag or system property to ignore is set to true.
+    */
+   private fun allowPrivate(): Boolean {
+      return context.configuration.ignorePrivateClasses ||
```

Comment:
Ignore seems inverted here too

Reply:
I've gone with ignorePrivateClasses in this revision.
Since include is now the default.
Just pushed some changes + test.